### PR TITLE
vjudge: add yacs

### PR DESCRIPTION
**Pull request by Molmin**
顺手修了 JudgeHandler next 函数的一些小问题，并为 vjudge 包适配新 api

### CODE REVIEW

File: packages/vjudge/src/providers/yacs.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    正在评测: STATUS.STATUS_JUDGING,
+    答案正确: STATUS.STATUS_ACCEPTED,
+    编译失败: STATUS.STATUS_COMPILE_ERROR,
+    答案错误: STATUS.STATUS_WRONG_ANSWER,
+    部分正确: STATUS.STATUS_WRONG_ANSWER,
+    运行时错误: STATUS.STATUS_RUNTIME_ERROR,
+    运行超时: STATUS.STATUS_TIME_LIMIT_EXCEEDED,
+    内存超出: STATUS.STATUS_MEMORY_LIMIT_EXCEEDED,
+    暂未公布: STATUS.STATUS_SYSTEM_ERROR,
+    评测机故障: STATUS.STATUS_SYSTEM_ERROR,
```
Comment (undefined-moe) [inline packages/vjudge/src/providers/yacs.ts]: move these to ../verdict

File: packages/vjudge/src/providers/yacs.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    async getProblem(id: string) {
+        logger.info(id);
+        const { problem } = await this.getData(`/problem/${id.split('P')[1]}`);
+        const tag = [];
+        if (problem.level) tag.push(problem.level);
+        if (problem.contest) tag.push(problem.contest.name);
+        let content = '';
+        if (problem.background.trim()) content += `## 题目背景\n\n${problem.background.trim()}\n\n`;
```
Comment (undefined-moe) [inline packages/vjudge/src/providers/yacs.ts:55]: i18n preferred

File: packages/vjudge/src/providers/yacs.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        const data = await this.getData(`/problem?pi=${page}`);
+        return data.problemList
+            .filter((problem) => problem.contest.status === '榜单已公布')
+            .map((problem) => `P${problem.id}`);
+    }
+
+    async submitProblem(id: string, lang: string, code: string, info, next, end) {
+        const langs = {
+            'yacs.1': 'C++',
+            'yacs.2': 'Python 3.6',
```
Comment (undefined-moe) [inline packages/vjudge/src/providers/yacs.ts]: using a seperate scope is no longer recommended. please use `validAs` property.

https://github.com/hydro-dev/Hydro/blob/master/packages/vjudge/src/model.ts#L50

```
langs:
  cc:
    compile: g++ ...
    execute: /w/code
    validAs:
      yacs: C++
    ...
```

while the new format provides additional features like pretest.

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/vjudge/src/providers/yacs.ts (hunk 2 above)
Comment (Molmin) [reply packages/vjudge/src/providers/yacs.ts:55]: > i18n preferred

这个网站全部是中文题目，感觉没必要用 i18n 吧？

另外怎么在后端用 i18n（）

File: packages/vjudge/src/verdict.ts [hunk 4]
```diff
@@ -39,6 +39,18 @@ export const VERDICT = new Proxy<Record<string, STATUS>>({
     SUBMIT_FAILED_PERM: STATUS.STATUS_SYSTEM_ERROR,
     SUBMIT_FAILED_TEMP: STATUS.STATUS_SYSTEM_ERROR,
 
+    // YACS
+    正在评测: STATUS.STATUS_JUDGING,
+    答案正确: STATUS.STATUS_ACCEPTED,
+    编译失败: STATUS.STATUS_COMPILE_ERROR,
+    // 答案错误: STATUS.STATUS_WRONG_ANSWER,
```
Comment (undefined-moe) [inline packages/vjudge/src/verdict.ts]: remove comments here

**Review by undefined-moe — APPROVED** [review]

**Pull request merged**